packages/language-server/src/server.ts

```diff
@@ -45,7 +49,18 @@ export function startServer(options?: LSOptions) {
   const connection: IConnection = getConnection(options)
   const documents: TextDocuments<TextDocument> = new TextDocuments(TextDocument)
 
-  connection.onInitialize(async () => {
+  // Does the clients accepts diagnostics with related information?
+  let hasCodeActionLiteralsCapability: boolean = false
+
+  connection.onInitialize(async (params: InitializeParams) => {
+    const capabilities = params.capabilities
+
+    hasCodeActionLiteralsCapability = !!(
+      capabilities.textDocument &&
+      capabilities.textDocument.codeAction &&
+      capabilities.textDocument.codeAction.codeActionLiteralSupport
+    )
```

Comment:
A better way to write this would be `Boolean(capabilities?.textDocument?.codeAction?.codeActionLiteralSupport)`